src/controllers/notificationController.js

```diff
@@ -0,0 +1,50 @@
+import Response from '../helpers/sendResponse';
+import code from '../helpers/statusCode';
+import notificationService from '../services/notificationServices';
+
+/** Class representing trip controller */
+class Notifications {
+  /**
+* @description this method gets all notifications
+* @param {Object} req request
+* @param {Object} res response
+* @return {object} Oject of data or error
+* @memberof commentController
+*/
+  static async getNotifications(req, res) {
+    try {
+      const { id } = req.user;
+      const data = await notificationService.getNotifications({ receiverId: id });
+      return Response.success(
+        res,
+        code.ok,
+        'Your Notifications have been retrieved successfully',
+        data
+      );
+    } catch (error) {
+      return Response.error(res, code.serverError, 'Oops something went wrong!');
+    }
+  }
+
+  /**
+* @description this method marks one or all as read
+* @param {Object} req request
+* @param {Object} res response
+* @return {object} Oject of data or error
+* @memberof commentController
+*/
+  static async markAsRead(req, res) {
+    try {
+      const param = req.query.id ? { id: req.query.id } : { receiverId: req.user.id };
```

Comment:
Can you shed more light on this logic? @James-nyagatare

Reply:
The reason I used this way it's because I am using this controller for both mark all as read and mark single as read with mark all as read it will be checking the receiverId in the notifications table while with mark one as Read it will be checking the id of the single notification in Notification model and that req.query.id, I have used because in the route I wrote localhost:5000/api/v1/notification/read?id=2 that id after a question mark is what we receive as req.query in the controller.